tripal/includes/TripalImporter.inc

```diff
@@ -0,0 +1,775 @@
+        $arguments['files'][] = [
+          'file_local' => $file_details['file_local'],
+          'file_path' => $file_details['file_local'],
+        ];
+        $has_file++;
+      }
+      if (array_key_exists('file_remote', $file_details)) {
+        $arguments['files'][] = [
+          'file_remote' => $file_details['file_remote'],
+        ];
+        $has_file++;
+      }
+      if (array_key_exists('fid', $file_details)) {
+        $values['fid'] = $file_details['fid'];
+        // Handle multiple file uploads.
+        if (preg_match('/\|/', $file_details['fid'])) {
+          $fids = explode('|', $file_details['fid']);
+          foreach ($fids as $fid) {
+            $file = file_load($fid);
+            $arguments['files'][] = [
+              // 'file_path' => drupal_realpath($file->uri), // old
+              'file_path' => \Drupal::service('file_system')->realpath($file->getFileUri()),
+              'fid' => $fid,
+            ];
+            $has_file++;
+          }
+        }
+        // Handle a single file.
+        else {
+          $fid = $file_details['fid'];
+          $file = file_load($fid);
+          $arguments['files'][] = [
+            // 'file_path' => drupal_realpath($file->uri), // old
+            'file_path' => \Drupal::service('file_system')->realpath($file->getFileUri()),
+            'fid' => $fid,
+          ];
+          $has_file++;
+
+          // For backwards compatibility add the old 'file' element.
+          $arguments['file'] = [
+            // 'file_path' => drupal_realpath($file->uri), // old
+            'file_path' => \Drupal::service('file_system')->realpath($file->getFileUri()),
+            'fid' => $fid,
+          ];
+        }
+      }
+
+      // Validate the $file_details argument.
+      if ($has_file == 0 and $class::$file_required == TRUE) {
+        throw new Exception("Must provide a proper file identifier for the \$file_details argument.");
+      }
+
+      // Store the arguments in the class and serialize for table insertion.
+      $this->arguments = $arguments;
+      //$values['arguments'] = serialize($arguments); //old serialize arguments
+      $values['arguments'] = base64_encode(serialize($arguments));
+
+      // Temporary debug code
+      // $logger = \Drupal::service('tripal.logger');
+      $this->logger->notice(
+        "arguments:\n" . print_r($arguments, true) . "\n\n"
+      );
+
+      // Insert the importer record.
+      $import_id = db_insert('tripal_import')
+        ->fields($values)
+        ->execute();
+
+      $this->import_id = $import_id;
+    } catch (Exception $e) {
+      throw new Exception('Cannot create importer: ' . $e->getMessage());
+    }
+  }
+
+  /**
+   * Loads an existing import record into this object.
+   *
+   * @param $import_id
+   *   The ID of the import record.
+   */
+  public function load($import_id) {
+    $class = get_called_class();
+
+    // Get the importer.
+    $import = db_select('tripal_import', 'ti')

```

Comment:
See deprecation comment above. Change as mentioned in https://www.drupal.org/node/2993033